File: python/adaptador/AudioPlayer.py
from adapter.MediaPlayer import MediaPlayer
from adapter.MP3Adapter import MP3Adapter
from adapter.MP4Adapter import MP4Adapter
from adapter.VLCAdapter import VLCAdapter


class AudioPlayer():
    def __init__(self):
        # Registra os adaptadores em um dicionário
        self.adapter_registry = {
            "mp3": MP3Adapter(),
            "mp4": MP4Adapter(),
            "vlc": VLCAdapter()
        }

    def play(self, audio_type, file_name):
        # Recupera o adaptador apropriado pelo tipo de áudio
        adapter = self.adapter_registry.get(audio_type.lower())
        if adapter:
            adapter.play(audio_type, file_name)
        else:
            print(f"Invalid media type. {audio_type} format not supported.")

File: java/adaptador/midia-player/src/main/java/edu/uea/AudioPlayer.java
package edu.uea;

import java.util.HashMap;
import java.util.Map;

import edu.uea.adapter.MP3Adapter;
import edu.uea.adapter.MP4Adapter;
import edu.uea.adapter.MediaPlayer;
import edu.uea.adapter.VLCAdapter;

// Client Class
public class AudioPlayer {
    private final Map<String, MediaPlayer> adapterRegistry = new HashMap<>();

    // Registra os adaptadores para os tipos de mídia suportados
    public AudioPlayer() {
        adapterRegistry.put("mp3", new MP3Adapter());
        adapterRegistry.put("mp4", new MP4Adapter());
        adapterRegistry.put("vlc", new VLCAdapter());
    }

    // Toca o arquivo de mídia usando o adaptador apropriado
    public void play(String audioType, String fileName) {
        MediaPlayer adapter = adapterRegistry.get(audioType.toLowerCase());
        if (adapter != null) {
            adapter.play(audioType,fileName);
        } else {
            System.out.println("Invalid media type. " + audioType + " format not supported.");
        }
    }
}